crates/starknet-devnet-server/src/restrictive_methods.rs

```diff
@@ -0,0 +1,119 @@
+use std::collections::HashMap;
+
+use lazy_static::lazy_static;
+lazy_static! {
+    static ref MAPPING_HTTP_URIS_TO_RPC_METHODS: HashMap<&'static str, &'static str> =
+        HashMap::from([
+            ("/dump", "devnet_dump"),
+            ("/load", "devnet_load"),
+            ("/set_time", "devnet_setTime"),
+            ("/increase_time", "devnet_increaseTime"),
+            ("/create_block", "devnet_createBlock"),
+            ("/abort_blocks", "devnet_abortBlocks"),
+            ("/restart", "devnet_restart"),
+            ("/mint", "devnet_mint"),
+            ("/postman/load_l1_messaging_contract", "devnet_postmanLoad"),
+            ("/postman/flush", "devnet_postmanFlush"),
+            ("/postman/send_message_to_l2", "devnet_postmanSendMessageToL2"),
+            ("/postman/consume_message_from_l2", "devnet_postmanConsumeMessageFromL2"),
+            ("/predeployed_accounts", "devnet_getPredeployedAccounts"),
+            ("/account_balance", "devnet_getAccountBalance"),
+            ("/config", "devnet_getConfig"),
+        ]);
+    static ref MAPPING_RPC_METHODS_TO_HTTP_URIS: HashMap<&'static str, &'static str> =
+        MAPPING_HTTP_URIS_TO_RPC_METHODS.iter().map(|(k, v)| (*v, *k)).collect();
+    pub static ref DEFAULT_RESTRICTED_JSON_RPC_METHODS: Vec<&'static str> = vec![
+        "devnet_mint",
+        "devnet_restart",
+        "devnet_createBlock",
+        "devnet_abortBlocks",
+        "devnet_impersonateAccount",
+        "devnet_autoImpersonate",
+        "devnet_getPredeployedAccounts"
```

Comment:
So now when a new method is added which needs to be restricted, what does the developer need to do? Populate this vector, add an entry to the mapping above? We need a list for this, perhaps in CONTRIBUTING.md. The list should actually contain general info regarding new method addition, which should also mention what needs to be done regarding dumping/loading. If you want, we can deal with this in a separate issue, but we must not forget it.